libraries/src/main/java/com/baeldung/smack/StanzaThread.java

```diff
@@ -0,0 +1,57 @@
+package com.baeldung.smack;
+
+import org.jivesoftware.smack.AbstractXMPPConnection;
+import org.jivesoftware.smack.SmackException;
+import org.jivesoftware.smack.XMPPException;
+import org.jivesoftware.smack.chat2.Chat;
+import org.jivesoftware.smack.chat2.ChatManager;
+import org.jivesoftware.smack.packet.Message;
+import org.jivesoftware.smack.tcp.XMPPTCPConnection;
+import org.jivesoftware.smack.tcp.XMPPTCPConnectionConfiguration;
+import org.jxmpp.jid.impl.JidCreate;
+import org.jxmpp.stringprep.XmppStringprepException;
+
+import java.io.IOException;
+
+public class StanzaThread extends Thread{
+
+
+    public StanzaThread(){
+    }
+
+    @Override
+    public void run() {
+        XMPPTCPConnectionConfiguration config = null;
+        try {
+            config = XMPPTCPConnectionConfiguration.builder()
+                    .setUsernameAndPassword("baeldung2","baeldung2")
+                    .setXmppDomain("jabb3r.org")
+                    .setHost("jabb3r.org")
+                    .build();
+
+            AbstractXMPPConnection connection = new XMPPTCPConnection(config);
+            connection.connect();
+            connection.login();
+
+
+            ChatManager chatManager = ChatManager.getInstanceFor(connection);
+
+            Chat chat = chatManager.chatWith(JidCreate.from("baeldung@jabb3r.org").asEntityBareJidOrThrow());
+
+            chat.send("Hello!");
+
+        } catch (XmppStringprepException e) {
+            e.printStackTrace();
```

Comment:
Let's not print out the stack trace. Instead I recommend either logging it or rethrowing an `IllegalArgumentException` if it is the user's fault (bad input, etc.) or an `IllegalStateException` if it is the environment's fault.